solidity/contracts/Gravity.sol

```diff
@@ -35,19 +35,20 @@ contract Gravity is ReentrancyGuard, Pausable {
 	// These are updated often
 	bytes32 public state_lastValsetCheckpoint;
 	mapping(address => uint256) public state_lastBatchNonces;
-	uint256 public state_lastValsetNonce = 0;
+	mapping(bytes32 => uint256) public state_invalidationMapping;
+	uint256 public state_lastValsetNonce;
 	// event nonce zero is reserved by the Cosmos module as a special
 	// value indicating that no events have yet been submitted
 	uint256 public state_lastEventNonce = 1;
 
 	// These are set once at initialization
 	bytes32 public state_gravityId;
```

Comment:
shouldn't this be immutable also?

Reply:
Correct, state_gravityId can be made immutable as well.